src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -163,11 +164,21 @@ public function loadClass($class)
                 throw new \RuntimeException(sprintf('Case mismatch between loaded and declared class names: %s vs %s', $class, $name));
             }
 
+            if (preg_match('#\n \* @final(?:( .+?)\.?)?\r?\n \*(?: @|/$)#s', $refl->getDocComment(), $notice)) {
+                self::$final[$name] = isset($notice[1]) ? preg_replace('#\s*\r?\n \* +#', ' ', $notice[1]) : '';
+            }
+
+            $parent = get_parent_class($class);
+            if ($parent && isset(self::$final[$parent])) {
+                @trigger_error(sprintf('The %s class is considered final%s. It may change without further notice as of its next major version. You should not extend it from %s.', $parent, self::$final[$parent], $name), E_USER_DEPRECATED);
```

Comment:
So.
If the class is `@final`, it means that one may not extend it, but still could use it.
If the class is `@internal`, it means that he should not use it at all.

For me it's more like that case:
```diff
-'The %s class is considered final%s. It may change without further notice as of its next major version. You should not extend it from %s.'
+'The %s class is considered final%s. It will be declared with `final` keyword in next major release. You should not extend it from %s.'
```

If one would like to ask user to not use that class (as he cannot rely on it's interface), I would rather mark that classes as `@internal` ones.